calcom/cal.com
# fix: #16083. booking location editable

## What does this PR do?

`if (!!val.credentialId)` it fails with `0` which is (Cal.com location). 
I've changed with a `typeof and >= 0`
There is an issue described in #16141. Some providers, like Google meet, have teamName = null. 

- Fixes #16083 
- Fixes CAL-4116 

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

- Are there environment variables that should be set? regular setup
- What are the minimal test data to have? I use data from pro@example.com user
- What is expected (happy path) to have (input and output)? update the booking location
- Any other important info that could help to test that PR
